kong/plugins/cors/handler.lua

```diff
@@ -7,12 +7,24 @@ CorsHandler.PRIORITY = 2000
 
 local OPTIONS = "OPTIONS"
 
+local function host_in_domain(domain, conf)
+  for i, d in ipairs(conf.origin) do
+    if string.match(domain, '[%.|//]'..d..'$') then
+      return true
+    end
+  end
+  return false
+end
+
 local function configure_origin(ngx, conf)
+  local origin = ngx.req.get_headers()["origin"] or nil
```

Comment:
let's only do this in the `else` branch a couple lines underneath. Also, no need for `or nil`.